src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,202 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private Node<T> currentNode;
+    private Node<T> previousNode;
+    private Node<T> nextNode;
+    private int size;
+
     @Override
     public void add(T value) {
+        if (size == 0) {
+            Node<T> newElement = new Node<T>(value);
+            head = newElement;
+            tail = newElement;
+        } else {
+            Node<T> newElement = new Node<T>(value);
+            newElement.prev = tail;
+            Node<T> previousNode = tail;
+            previousNode.next = newElement;
+            tail = newElement;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException("The index passed must be greater or equal to 0 "
+                    + "and less or equal to" + size);
+        }
+        if (size - index == 0 || size == 0) {
+            add(value);
+            return;
+        }
+        iterateCurrentNode(index);
+        addNodeInTheMiddle(value);
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        verifyIndex(index);
+        iterateCurrentNode(index);
+        return currentNode.item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        if (size == 0) {
+            throw new IndexOutOfBoundsException("The list is empty");
+        }
+        verifyIndex(index);
+        iterateCurrentNode(index);
+        T returnItem = currentNode.item;
+        currentNode.item = value;
+        return returnItem;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        verifyIndex(index);
+        if (index == 0) {
+            return removeFirstNode();
+        }
+        iterateCurrentNode(index);
+        assignPreviousAndNextNode();
```

Comment:
The method 'assignPreviousAndNextNode' seems to be redundant. Consider removing it if it's not necessary for your implementation.